src/include/stir/scatter/ScatterSimulation.h

```diff
@@ -0,0 +1,433 @@
+/*
+    Copyright (C) 2004 - 2009 Hammersmith Imanet Ltd
+    Copyright (C) 2013 - 2016 University College London
+    This file is part of STIR.
+
+    This file is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.
+
+    This file is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    See STIR/LICENSE.txt for details
+*/
+#ifndef __stir_scatter_ScatterSimulation_H__
+#define __stir_scatter_ScatterSimulation_H__
+
+/*!
+  \file
+  \ingroup scatter
+  \brief Definition of class stir::ScatterSimulation
+
+  \author Nikos Efthimiou
+  \author Kris Thielemans
+*/
+
+#include "stir/ParsingObject.h"
+#include "stir/RegisteredObject.h"
+#include "stir/ProjData.h"
+#include "stir/VoxelsOnCartesianGrid.h"
+#include "stir/ProjDataInfoCylindricalNoArcCorr.h"
+
+START_NAMESPACE_STIR
+
+/*!
+  \ingroup scatter
+  \brief Simuate the scatter probability using a model-based approach
+
+  N.E. : This class is roughly the base class of what used to be the ScatterEstimationByBin.
+  Because there are different approaches on the actual simulation process, this base class will
+  be in charge of hold projection data and downsample the attenuation image, while more core function
+  will deligate to classes like SingleScatterSimulation.
+
+  This class computes the single Compton scatter estimate for PET data using an analytical
+  approximation of an integral. It takes as input an emission image and an attenuation image.
+  This is effectively an implementation of the simulation part of the algorithms of
+  Watson and Ollinger.
+
+  One non-standard feature is that you can specify a different attenuation image to find the
+  scatter points and one to compute the integrals over the attenuation image. The idea is that
+  maybe you want to compute the integrals on a finer grid than you sample the attenuation image.
+  This is probably not very useful though.
+
+  \todo Currently this can only be run by initialising it via parsing of a file. We need
+  to add a lot of set/get members.
+
+  \todo This class currently uses a simple Gaussian model for the energy resolution. This model
+  and its parameters (\a reference_energy and \a energy_resolution) really should be moved the
+  the Scanner class. Also the \a lower_energy_threshold and \a upper_energy_threshold should
```

Comment:
so, let's not set them here!

Reply:
This has been done. Removing the comment.